pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -703,8 +708,10 @@ stdenv.mkDerivation (finalAttrs: {
     '';
     license = with licenses; [ lgpl21Plus ]
```

Comment:
I don't know how license works (IANAL), but it looks a bit self-contradictory to include free licenses with unfree ones.
It should be something like "if unfree components were selected then license is unfree; else, insert free licenses accordingly".

Reply:
I think its still useful to know if there is lgpl/gpl code included in a binary even if the resulting binary is unfree (which is not even true in all cases where the unfree option is enabled)

Reply:
The free licenses also all still apply to at least some parts of the code. Distributors must accompany the artifacts with the license text and copyright notices with most FOSS licenses. IANAL but I think we're fulfilling our legal obligations here.

The real "license" of a package is also always the composite of all of these licenses. If you don't accept one of them, you don't even have the right to download and run the package.  

This nature of licenses is also important in cases such as i.e. `[ unfreeRedistributable gpl3Plus ]` where you *cannot* actually redistribute the resulting package as the GPLv3 only allows redistribution according to its terms for the entire code-base which are not compatible with the terms of unfreeRedistributable.

We need a proper system for these composite license due to problems like this. I'll open an issue on that.

Reply:
Forgot to add a back link: https://github.com/NixOS/nixpkgs/issues/283706